### PR TITLE
WIP: added delete button

### PR DESCRIPTION
**Pull request by JoHeyJo**
Added deleteProduct()  & button to delete products from database.

### CODE REVIEW

**Comment by lananelson** [comment]
thank you for exploring what this might look like! since deleting a product is a less common action (more common would be for example to mark a product is out of stock) we should think about how to enable less common actions. One common pattern (used in google drive) is to leverage check boxes https://tailwindui.com/components/application-ui/lists/tables, see this screenshot
image

**Comment by lananelson** [comment]
it would be awesome if you could add checkboxes and delete action! then later we can add buttons for other actions, e.g. "Mark as Out of Stock" which will update the product setting quantity to 0

**Comment by JoHeyJo** [comment]
Sounds good! Yeah, that makes sense a lot of sense from a UX standpoint. I like it.

**Comment by JoHeyJo** [comment]
just a thought - for the less common actions, like 'delete' maybe that can reside in the form?

File: pages/pantries/[id].tsx [hunk 1]
```diff
@@ -115,11 +130,41 @@ export default function Pantry() {
...
+    console.log('category.products',category.products)
+    if(category.products){
+      category.products.map((product) => {
+       console.log('acc',acc)
+       acc.push(product);
+      })
+    }
+    return acc
+  },[])
+
```
Comment (JoHeyJo) [inline pages/pantries/[id].tsx:152]: I had to massage some data to apply the logic from the template.

File: components/Checkbox.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
+/*
```
Comment (JoHeyJo) [inline components/Checkbox.tsx:1]: This whole thing can be deleted, I just kept it as a reference for the time being. Unless we do decide to make Checkbox its own component.

File: pages/pantries/[id].tsx (hunk 1 above)
Comment (lananelson) [reply pages/pantries/[id].tsx:194]: hmm why not `products = pantry.products`?

**Comment by lananelson** [comment]
a couple of notes on the visuals:
1. the Select all checkbox should be moved to the top row where all of the actions and filters are. Users will look there for actions since we are putting all the buttons there
2. there's background gray behind bulk actions and white background in part of the category on the right side which needs to be removed
3. we should keep the vertical alignment of Table headers, Category name, and Product names. It makes the table more scannable. I tried to illustrate below (it is not pixel perfect, still use your judgement and eye please for the final result)
image

**Comment by lananelson** [comment]
I could not get the delete to work when clicking "delete all" button

File: pages/pantries/[id].tsx [hunk 3]
```diff
@@ -199,11 +235,36 @@ export default function Pantry() {
...
+                      {pantry.products.length === selectedProduct.length ? "Delete all" : "Delete"}
+                    </button>
+                  </div>
+                )}
                 <Filter validCategories={categories.filter(category => category.products)} />
                 <table className="min-w-full">
                   <thead className="bg-white">
                     <tr>
+                      <th scope="col" className="relative w-12 px-6 sm:w-16 sm:px-8">
+                      </th>
```
Comment (JoHeyJo) [inline pages/pantries/[id].tsx:267]: Adding this `<th>` aligns everything really nicely. Not sure if it's 'hacky' or not bc it's empty but if I tried to adjust it with styling/css it just didn't align as well, also it didn't size properly.

File: pages/pantries/[id].tsx (hunk 1 above)
Comment (lananelson) [reply pages/pantries/[id].tsx]: lol yeppp

File: components/Checkbox.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  // More people...
+]
+
+function classNames(...classes) {
+  return classes.filter(Boolean).join(' ')
+}
+//when i check name check box
+export default function Checkbox () {
+  const checkbox = useRef()
+  const [checked, setChecked] = useState(false) // true
```
Comment (lananelson) [inline components/Checkbox.tsx:19]: nit: remove the comments that you don't need any more

File: components/Checkbox.tsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  const [indeterminate, setIndeterminate] = useState(false)// false
+  const [selectedPeople, setSelectedPeople] = useState([]) //{email: "lindsay.walton@example.com", name: "Lindsay Walton", title: "Front-end Developer", role: "Member"  }
+
+  useLayoutEffect(() => {
+    const isIndeterminate = selectedPeople.length > 0 && selectedPeople.length < people.length
```
Comment (lananelson) [inline components/Checkbox.tsx:24]: Need to rename the variables from people to products

File: pages/pantries/[id].tsx [hunk 6]
```diff
@@ -133,6 +138,20 @@ export default function Pantry() {
...
   }
 
+  /** removes selected product(s) from database and rerenders updated category list */
+  async function deleteProduct() {
+    for (let product of selectedProduct) {
+      const { data, error } = await supabase
+        .from('products')
+        .delete()
+        .eq('id', product.id)
+        .single();
```
Comment (lananelson) [inline pages/pantries/[id].tsx:148]: hmmm I wonder if this `.single()` can be leveraged somehow to not run a bunch of separate queries and instead run one delete query for all products... Have you looked into it?

File: pages/pantries/[id].tsx [hunk 7]
```diff
@@ -162,6 +181,23 @@ export default function Pantry() {
...
   }
 
+  // ################ checkbox logic ##################
+
+  // useLayoutEffect(() => {
+  //   const isIndeterminate = selectedProduct.length > 0 && selectedProduct.length < categoriesWithProducts.products.length
+  //   setChecked(selectedProduct.length === categoriesWithProducts.products.length)
+  //   setIndeterminate(isIndeterminate)
+  //   checkbox.current.indeterminate = isIndeterminate
+  // }, [selectedProduct])
```
Comment (lananelson) [inline pages/pantries/[id].tsx:191]: nit: remove ununsed comments

File: pages/pantries/[id].tsx (hunk 3 above)
Comment (lananelson) [reply pages/pantries/[id].tsx:267]: that makes sense and empty cells is a norm in Excell tables so I'd say not a hack!

**Comment by lananelson** [comment]
Looking very sleek! There is a small edge cases to fix:
- negative increment of quantity is possible

File: pages/pantries/[id].tsx (hunk 6 above)
Comment (lananelson) [reply pages/pantries/[id].tsx:147]: can use `in` operator but it has limitations https://github.com/supabase/postgrest-js/issues/292

**Pull request merged**